test/WebJobs.Script.Tests/ScriptHostTests.cs

```diff
@@ -451,7 +451,7 @@ public void ApplyConfiguration_Queues()
             scriptConfig.HostConfig.HostConfigMetadata = config;
             TraceWriter traceWriter = new TestTraceWriter(TraceLevel.Verbose);
 
-            scriptConfig.HostConfig.CreateMetadataProvider(); // will cause extensions to initialize and consume config metadata.
+            new JobHost(scriptConfig.HostConfig).CreateMetadataProvider(); // will cause extensions to initialize and consume config metadata.
```

Comment:
There are two sets of changes in the PR, not much context on the second one. Can you provide more information o this?